src/bun.js/ipc.zig

```diff
@@ -194,21 +194,29 @@ const json = struct {
     // 1 is regular
     // 2 is internal
 
-    pub fn decodeIPCMessage(
-        data: []const u8,
-        globalThis: *JSC.JSGlobalObject,
-    ) IPCDecodeError!DecodeIPCMessageResult {
+    pub fn decodeIPCMessage(data: []const u8, globalThis: *JSC.JSGlobalObject) IPCDecodeError!DecodeIPCMessageResult {
         if (bun.strings.indexOfChar(data, '\n')) |idx| {
             var kind = data[0];
-            var json_data = data[1..idx];
+            var json_data = data[0..idx];
 
             switch (kind) {
-                1, 2 => {},
-                else => {
-                    // if the message being recieved is from a node process then it wont have the leading marker byte
-                    // assume full message will be json
+                2 => {
+                    json_data = data[1..idx];
+                },
+                '{', //object
+                '[', //array
+                '"', //string
+                '0'...'9', //number
+                't', //true
+                'f', //false
+                'n', //null
+                => {
                     kind = 1;
-                    json_data = data[0..idx];
+                },
+                else => {
+                    std.log.warn("{d} {d} {d}", .{ kind, json_data.len, idx });
+                    std.log.warn("{d}", .{json_data});
+                    @panic("ipc: invalid json");
```

Comment:
This shouldn't hard crash, it should do what node does (emit an exception event, close the socket, allow the process to exit due to the unhandled exception if necessary)